core/services/versionchooser/versionchooserutils/chooser.py

```diff
@@ -0,0 +1,163 @@
+            web.StreamResponse: Streams the 'docker pull' output
+        """
+        response = web.StreamResponse()
+        response.headers["Content-Type"] = "application/x-www-form-urlencoded"
+        # This step actually starts the chunked response
+        await response.prepare(request)
+
+        low_level_api = docker.APIClient(base_url="unix://var/run/docker.sock")
+        # Stream every line of the output back to the client
+        for line in low_level_api.pull(f"{repository}:{tag}", stream=True, decode=True):
+            await response.write(f"{line}\n\n".replace("'", '"').encode("utf-8"))
+        await response.write_eof()
+        return response
+
+    async def set_version(self, image: str, tag: str) -> web.StreamResponse:
+        """Sets the current version.
+
+        Sets the version in startup.json()
+
+        Args:
+            image (str): the repository of the image
+            tag (str): the desired tag
+
+        Returns:
+            web.Response:
+                200 - OK
+                400 - Invalid image/tag
+                500 - Invalid settings file/Other internal error
+        """
+        if not self.is_valid_version(image, tag):
+            return web.Response(status=400, text="Invalid version")
+
+        with open(DOCKER_CONFIG_PATH, "r+") as startup_file:
+            try:
+                data = json.load(startup_file)
+                data["core"]["image"] = image
+                data["core"]["tag"] = tag
+
+                # overwrite file contents
+                startup_file.seek(0)
+                startup_file.write(json.dumps(data, indent=2))
+                startup_file.truncate()
+
+                logging.info("Starting bootstrap...")
+                bootstrap = self.client.containers.get("companion-bootstrap")
+                bootstrap.start()
+
+                logging.info("Stopping core...")
+                core = self.client.containers.get("companion-core")
+                core.kill()
+                return web.Response(status=200, text=f"Changed to version {image}:{tag}, restarting...")
+
+            except KeyError:
+                return web.Response(status=500, text="Invalid version file")
+
+            except Exception as error:
+                logging.critical("Error: %s: %s", type(error), error)
+                return web.Response(status=500, text=f"Error: {type(error)}: {error}")
+
+    async def get_available_versions(self, repository: str, tag: str) -> web.Response:
+        """Returns versions available locally and in the remote
+
+        Args:
+            repository (str): repository name (such as bluerobotics/companion-core)
+            tag (str): tag (such as "master" or "latest")
+
+        Returns:
+            web.Response: json described in the openapi file
+        """
+        output: Dict[str, List[Dict[str, str]]] = {"local": [], "remote": []}
+        image_name = f"{repository}/{tag}"
+        for image in self.client.images.list():
```

Comment:
`filter(lambda image: "companion-core" not in image.tags, images)`

Reply:
image.tags is a list, we need t check each of the tags.

I could do `filter(lambda image: any("companion-core" in tag for tag in image.tags),  images)`
but that is not nice to read